iid/iid.go

```diff
@@ -84,8 +146,11 @@ func (c *Client) DeleteInstanceID(ctx context.Context, iid string) error {
 		return err
 	}
 
-	if msg, ok := errorCodes[resp.Status]; ok {
-		return fmt.Errorf("instance id %q: %s", iid, msg)
+	if info, ok := errorCodes[resp.Status]; ok {
```

Comment:
I notice we're only using the http status code here. Can't our APIs return more specific error codes in the body of the response? Or am I thinking of something else?

Reply:
The endpoint we use for this functionality today signals errors via HTTP status code. In the future the backend service will send more detailed and well-formed error responses.